### PR TITLE
Promotions types

### PR DESCRIPTION
**Pull request by pannga**

### CODE REVIEW

**Comment by pannga** [comment]
I think this one is easiest to merge && and will add 'export' to Vouchers Discount interfaces

File: packages/sdk/src/Promotions.ts [hunk 1]
```diff
@@ -7,13 +9,13 @@ export class Promotions {
 	/**
 	 * @see http://docs.voucherify.io/reference?utm_source=github&utm_medium=sdk&utm_campaign=acq#create-promotion-campaign
 	 */
-	public create(params: $FixMe) {
-		return this.campaignsNamespace.create(params)
+	public create(campaign: T.PromotionsCreate) {
+		return this.campaignsNamespace.create(campaign)
```
Comment (darekg11) [inline packages/sdk/src/Promotions.ts:13]: Please add typing

File: packages/sdk/src/types/Promotions.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	name: string
+	campaign_type: 'PROMOTION'
+	start_date?: string
+	expiration_date: string
+	promotion: {
+		tiers: {
+			name: string
+			banner: string
+			action: {
+				discount?: DiscountUnit | DiscountAmount | DiscountPercent
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:15]: Drop `?`

File: packages/sdk/src/types/Promotions.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		source_id?: string
+		items?: OrdersItem[]
+		amount?: number
+		metadata?: Record<string, any>
+	}
+	metadata?: Record<string, any>
+}
+
+export interface PromotionsValidateResponse {
+	valid?: boolean
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:34]: Drop `?`

File: packages/sdk/src/types/Promotions.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	promotions?: {
+		id?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:36]: Drop `?`

File: packages/sdk/src/types/Promotions.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+		object?: 'promotion_tier'
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:37]: Drop `?`

**Review by darekg11 — CHANGES_REQUESTED** [review]
Request changes: 
- types definitions
- please create promotion campaign with multiple different tiers and run validation with order and check response format

**Comment by pannga** [comment]
I've added typing to `create` method, but I had to do it manually - ive removed CampaignNamespace from Promotions.ts in order to add respone type (previously it was an `Expected 0 type arguments, but got 1.ts(2558)` error)

File: packages/sdk/src/types/Promotions.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import { DiscountUnit, DiscountAmount, DiscountPercent } from './Vouchers'
+import { SimpleCustomer } from './Customers'
+import { ValidationRulesCreateAssignmentResponse } from './ValidationRules'
+
+export interface PromotionsCreateResponse {
+	id: string
+	name: string
+	campaign_type?: 'PROMOTION'
+	type: 'AUTO_UPDATE' | 'STATIC'
+	description: string
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:11]: Are you sure that this is indeed always present in response?

File: packages/sdk/src/types/Promotions.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import { OrdersItem } from './Orders'
+import { DiscountUnit, DiscountAmount, DiscountPercent } from './Vouchers'
+import { SimpleCustomer } from './Customers'
+import { ValidationRulesCreateAssignmentResponse } from './ValidationRules'
+
+export interface PromotionsCreateResponse {
+	id: string
+	name: string
+	campaign_type?: 'PROMOTION'
+	type: 'AUTO_UPDATE' | 'STATIC'
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:10]: Only `STATIC` in case of promotions

**Review by darekg11 — CHANGES_REQUESTED** [review]
Just apply small fixes.

File: packages/sdk/src/types/Promotions.ts (hunk 6 above)
Comment (pannga) [reply packages/sdk/src/types/Promotions.ts:11]: `description` property is indeed in the promotions create response

**Comment by pannga** [comment]
Updated

**Comment by pannga** [comment]
Updated after sync call

File: packages/sdk/src/types/Promotions.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+export interface PromotionsCreate {
+	name: string
+	campaign_type: 'PROMOTION'
+	start_date?: string
+	expiration_date?: string
+	promotion?: {
+		tiers: {
+			name: string
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:77]: Please add `metadata?: Record<string, any>`

File: packages/sdk/src/types/Promotions.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+		items?: OrdersItem[]
+		amount?: number
+		metadata?: Record<string, any>
+	}
+	metadata?: Record<string, any>
+}
+
+export interface PromotionsValidateResponse {
+	valid: boolean
+	promotions?: {
```
Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:100]: Please add `metadata?: Record<string, any>`

Comment (darekg11) [inline packages/sdk/src/types/Promotions.ts:100]: If you send `order: { ... }` in request then this order will be also available in response so please include that

**Comment by pannga** [comment]
Updated as requested

**Review by darekg11 — APPROVED** [review]

**Pull request merged**